nautobot/core/celery/schedulers.py

```diff
@@ -0,0 +1,56 @@
+import logging
+
+from celery import current_app
+from django_celery_beat.schedulers import ModelEntry, DatabaseScheduler
+
+from nautobot.extras.models import ScheduledJob, ScheduledJobs
+
+
+logger = logging.getLogger(__name__)
+
+
+class NautobotScheduleEntry(ModelEntry):
+    """
+    Nautobot variant of the django-celery-beat ModelEntry which uses the
+    nautobot.extras.models.ScheduledJob model
+    """
+
+    def __init__(self, model, app=None):
+        """Initialize the model entry."""
```

Comment:
That’s a good question. @lampwins do you have any insight on that?

Reply:
@lampwins Can you comment on this? Are there parts of the base `ModelEntry.__init__()` that we're intentionally bypassing, and if so, why?

Reply:
Sorry for the delay.

It has been a while, but IIRC I did rip out a couple of model fields and thus there was a functionality that was no longer relevant or what wrong. Mostly around the use of rabbitmq I think.

Reply:
I had also changed the way the schedule period models work. Instead of separate models, I had baked it into the scheduled task model to simplify things, thus changing the logic.

Reply:
Good enough for me. Thanks for the context!